workers/unity/Assets/Gdk/Core/Exceptions/ConnectionFailedException.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+
+namespace Improbable.Gdk.Core
+{
+    public enum ConnectionErrorReason
+    {
+        CannotEstablishConnection,
+        DeploymentNotFound,
+        InvalidConfig
+    }
+
+    public class ConnectionFailedException : Exception
```

Comment:
Wouldn't it be useful to ensure the `ToString` of this exception includes the error reason?

Reply:
I don't think so, the messages are already sufficiently clear and different based on the reason. The aim of adding the Reason field was to make it easier to programmatically distinguish the errors.